libraries/AP_Baro/AP_Baro_BMP388.cpp

```diff
@@ -91,8 +93,10 @@ bool AP_Baro_BMP388::init()
     uint8_t whoami;
     if (!read_registers(BMP388_REG_ID, &whoami, 1)  ||
         whoami != BMP388_ID) {
-        // not a BMP388
-        return false;
+        if (whoami != BMP390_ID) {
```

Comment:
the nested if is a bit odd, I'd prefer a switch statement, and create a new DEVTYPE_BARO_BMP390 and use that in the dev->set_device_type() call. That way in a log we can see what sort of baro we have (it shows up in the "devid" command in MAVProxy for example, and in MAVExplorer)

Reply:
@tridge Okay I think I have all of your comments addressed and the commits renamed.